### PR TITLE
GO-3483: Upload image fallback to file upload if any processing

### PR DESCRIPTION
**Pull request by deff7**

### CODE REVIEW

File: core/files/fileobject/fileindex.go [hunk 1]
```diff
@@ -238,22 +238,30 @@ func (ind *indexer) buildDetails(ctx context.Context, id domain.FullFileId) (det
 	if err != nil {
 		return nil, "", err
 	}
-	if mill.IsImage(file.Info().Media) {
-		image, err := ind.fileService.ImageByHash(ctx, id)
+
+	if file.Info().Mill == mill.BlobId {
```
Comment (deff7) [inline core/files/fileobject/fileindex.go:242]: Test

File: core/files/fileobject/fileindex.go [hunk 2]
```diff
@@ -238,22 +238,30 @@ func (ind *indexer) buildDetails(ctx context.Context, id domain.FullFileId) (det
...
+		details, err = image.Details(ctx)
 		if err != nil {
 			return nil, "", err
 		}
 	}
+
+	// Overwrite typeKey for images in case that image is uploaded as file.
+	// That can be possible because some images can't be handled properly and wee fall back to
+	// handling them as files
+	if mill.IsImage(file.Info().Media) {
```
Comment (deff7) [inline core/files/fileobject/fileindex.go:261]: Test

File: core/files/fileuploader/uploader.go [hunk 3]
```diff
@@ -440,10 +446,16 @@ func (u *uploader) Upload(ctx context.Context) (result UploadResult) {
...
 	var addResult *files.AddResult
-	if u.fileType == model.BlockContentFile_Image {
+	if !u.forceUploadingAsFile && u.fileType == model.BlockContentFile_Image {
 		addResult, err = u.fileService.ImageAdd(ctx, u.spaceId, opts...)
-		if errors.Is(err, image.ErrFormat) || errors.Is(err, mill.ErrFormatSupportNotEnabled) {
-			return u.SetType(model.BlockContentFile_File).Upload(ctx)
+		if errors.Is(err, image.ErrFormat) ||
+			errors.Is(err, mill.ErrFormatSupportNotEnabled) ||
+			errors.Is(err, mill.ErrProcessing) {
+			err = nil
```
Comment (deff7) [inline core/files/fileuploader/uploader.go:454]: Test

File: core/files/image.go [hunk 4]
```diff
@@ -40,7 +40,7 @@ type image struct {
 func selectAndSortResizeVariants(variants []*storage.FileInfo) []*storage.FileInfo {
 	onlyResizeVariants := variants[:0]
 	for _, variant := range variants {
-		if variant.Mill == mill.ImageResizeId {
+		if variant.Mill == mill.ImageResizeId || variant.Mill == mill.BlobId {
```
Comment (deff7) [inline core/files/image.go:43]: Test

File: pkg/lib/gateway/gateway.go [hunk 5]
```diff
@@ -337,7 +337,7 @@ func (g *gateway) getImage(ctx context.Context, r *http.Request) (files.File, io
 			return nil, err
 		}
 		return res, nil
-	})
+	}, retryOptions...)
```
Comment (deff7) [inline pkg/lib/gateway/gateway.go:340]: Test

**Review by fat-fellow — APPROVED** [review]

**Pull request merged**